### ISSUE TITLE
Additional class names not added

### ISSUE THREAD

**Issue by stahlmanDesign** [nbudin/react-bootstrap4-modal#7]
Putting `dialogClassName` and `className` on the Modal component (next to visible and onClickBackdrop) does not add classes to the modal markup as described.

Using:
```
"bootstrap": "^4.0.0",
"react-bootstrap4-modal": "^1.2.0",
```

**Comment by vjau** [comment]
I think for some reasons, the code compiled by babel removes the additional class, hence 
```jsx
<div className={classNames('modal-dialog', dialogClassName)} role="document" onClick={this.stopPropagation}>
   <div className="modal-content">
              {children}
   </div>
</div>
```

becomes
```js
_react2.default.createElement(
            'div',
            { className: 'modal-dialog', role: 'document', onClick: _this.stopPropagation },
            _react2.default.createElement(
              'div',
              { className: 'modal-content' },
              _this.props.children
            )
          )
```
I don't know enough babel to fix it, waiting for @nbudin to chime in.

**Comment by stahlmanDesign** [comment]
I took a look at the classnames library here https://github.com/JedWatson/classnames but didn't see any clues.

Looking at the Modal compiled `render()` method shows it correctly calls the classnames function for the modal fade state just above the classnames call for dialogClassName (which doesn't compile as expected)

```
    _this.render = function () {
      return _react2.default.createElement(
        'div',
        null,
        _react2.default.createElement(
          'div',
          {
            className: (0, _classnames2.default)('modal', 'fade', { show: _this.state.visible }),
            style: { display: _this.state.visible || _this.state.transitioning ? 'block' : 'none' },
            role: 'dialog',
            'aria-hidden': !_this.state.visible,
            tabIndex: '-1',
            onClick: _this.props.onClickBackdrop
          },
          _react2.default.createElement(
            'div',
            { className: 'modal-dialog', role: 'document', onClick: _this.stopPropagation },
            _react2.default.createElement(
              'div',
              { className: 'modal-content' },
              _this.props.children
            )
          )
        ),
        _this.renderBackdrop()
      );
    };
```

I copied the /src folder into my project (which uses create-react-app and its dev server, Babel setup) and when using the uncompiled code everything works fine. So this does appear to be a bug with the transpiled code only.

**Comment by nbudin** [comment]
Whoa, this is bizarre.  Thanks for letting me know about it!  I'll take a look in a bit.

**Comment by stahlmanDesign** [comment]
I also use [Reactstrap](https://reactstrap.github.io) which has similar Boostrap 4 Modals, but with more options. Its code does add classes, so it could be worth looking how Babel is configured in that library.
Exampe:
```
<Modal
        isOpen={this.state.modal}
        toggle={this.toggle}
        className={ 'modal-dialog-wide' }>
…
```
Will add the custom class 'modal-dialog-wide' to 'modal-dialog'

**Comment by nbudin** [comment]
OK, the issue seems to be with my build toolchain, which I changed a bit for this release.  Previously, it was using Babel to compile stuff into `lib`, and Webpack to compile a minified and gzipped version into `dist`.  I decided to simplify things a bit and use Webpack for both in the 1.2.0 release, but unfortunately I forgot to change package.json so that it would be pointing at the Webpack-compiled version of things.  As a result, an old version of the code was leftover in `lib` and ended up released to npm.

I'm going to try to figure out how to get everything actually working ok with Webpack builds, and do a 1.2.1 release once it is.  Sorry for the hassle, folks!

**Comment by nbudin** [comment]
OK, several tries later, I think I've got this sorted.

Version 1.2.2 is out (yup, I ended up publishing a bad 1.2.1 build and then unpublishing it, and then realizing later I should have deprecated it), and it works for me and allows adding `dialogClassNames`.  It's now using Webpack for all the builds, with a library configuration, and a properly-externalized React.

Sorry again for the hassle.  I feel like I've learned a lot as an npm package maintainer during this cycle of releases, so hopefully it'll be smoother sailing from here out.
